### PR TITLE
handle compressed book (and file open errors)

### PR DESCRIPTION
**Pull request by borg323**

### CODE REVIEW

File: src/chess/pgn.h [hunk 1]
```diff
@@ -43,14 +45,31 @@ struct Opening {
...
 };
 
+inline bool GzGetLine(gzFile file, std::string& line) {
+  bool flag = false;
+  char s[200];
+  line.clear();
+  while (gzgets(file, s, sizeof(s))) {
+    flag = true;
+    line += s;
+    auto r = line.find_last_of('\n');
```
Comment (Tilps) [inline src/chess/pgn.h:55]: Any issue if someone tries to use a windows line ending pgn file on windows?

Comment (borg323) [reply src/chess/pgn.h:55]: I hope opening the file in "r" mode will take care of it - but a test would be nice.

Comment (borg323) [reply src/chess/pgn.h:55]: A quick test shows both line encodings seem to work fine under windows and linux.

**Comment by Tilps** [comment]
Still approved :)

**Pull request merged**